python/lib/evaluations/dmod/evaluations/specification/fields.py

```diff
@@ -239,8 +241,31 @@ def apply_configuration(
         template_manager: TemplateManager,
         decoder_type: typing.Type[json.JSONDecoder] = None
     ):
-        self.where = configuration.get("where", self.where)
-        self.datatype = configuration.get("datatype", self.datatype)
+        if "where" in configuration:
```

Comment:
I dont understand the full context of this, so take this comment with a grain of salt. Reading through this newly introduced section of code, this looks very similar to creating a `ValueSelector` instance and then updating `self` with the properties from the newly created instance. Would it be easier to just do that?

Reply:
The template values in the passed in configuration could be just about anything and could quite possibly *not* pass validations upon construction. A `ValueSelector` **requires** a valid "where" value to be constructed, but the given template might just be a dict with defined associated values or something.

Reply:
Thanks for the explanation!